### DISCUSSION TITLE
invalid output in pr body for multiple updates in a pnpm monorepo managed with changesets

### DISCUSSION THREAD

**Discussion by dominikg** [renovatebot/renovate#14458] [Request Help] [answered]
### How are you running Renovate?

WhiteSource Renovate hosted app on github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

see https://github.com/sveltejs/vite-plugin-svelte/pull/284

* note that in the package table `@sveltejs/adapter-auto` is a link, `@sveltejs/kit` is not.
* expanding `sveltejs/kit` in the release-notes section shows the changelog for adapter-auto 
 
both packages are hosted in the same repo: https://github.com/sveltejs/kit . Their package.json contains directory information for the package path inside the repo.

Is there something that can be done on the projects side to help renovate parse these correctly?

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste any log here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

No reproduction, but I have linked to a public repo where it occurs

**Comment by rarkins** [comment, marked as answer]
Hi @dominikg,

Please don't raise any more bug reports and always raise Discussions first, especially if you don't have logs or a reproduction. I have converted this one out of courtesy but it takes up my time and in future I will one-click delete instead.

Look at https://registry.npmjs.org/@sveltejs/adapter-auto. Then look at https://registry.npmjs.org/@sveltejs/kit. I recommend using Firefox as you'll get it nicely formatted. Notice the missing `repository` field in `@sveltejs/kit`. I'm not sure why it's missing from the npmjs registry for that package, but Renovate requires it to know the source URL.

I suggest you raise an issue with the Svelte repo in question.

**Comment by dominikg** [comment]
thanks. i'll look into it 

for the record:
I raised it as a bug because the release-notes section contains and invalid entry (it shows data for a different package). Even if a package doesn't contain all required metadata, it shouldn't display possibly misleading information from another package.

I did post a link to a public repository containing a PR with invalid data, which was an explicit select option in your bug template. If you always require a minimal reproduction i suggest you mark that field as required and remove that option.

**Reply by rarkins** [reply]
> for the record: I raised it as a bug because the release-notes section contains and invalid entry (it shows data for a different package). Even if a package doesn't contain all required metadata, it shouldn't display possibly misleading information from another package.

To make sure I'm not missing anything, can you tell me *exactly* what you mean by "data for a different package" / "misleading information from another package"? The sections I see are in this screenshot:

![image](https://user-images.githubusercontent.com/6311784/156044297-3c4abf55-9b8e-418e-9962-c2807e67d228.png)

They are all *repository* names, not package names (i.e. it means `github.com/sveltejs/git` and not `@sveltejs/git` on npm.

I'm pretty sure what you think is a bug is actually Renovate's intended behavior, where the `@sveltejs/kit` package has no `repository` field in its metadata on npm.

> I did post a link to a public repository containing a PR with invalid data, which was an explicit select option in your bug template. If you always require a minimal reproduction i suggest you mark that field as required and remove that option.

We don't always require it, but if we find that anyone regularly raises misunderstandings or questions as bugs, and makes little effort to verify them, then we do need to be more strict. To be frank I didn't notice that you're an open source contributor yourself, and probably wouldn't have said it if I had :)

**Reply by dominikg** [reply]
I'm sorry for not including more details in the initial report and it looks like i might have misunderstood the collapsibles under release-notes.
this section reads like it is the changelog for the package @sveltejs/kit 
![Screenshot_2022-02-28_20-32-49](https://user-images.githubusercontent.com/611613/156047133-ff1444dd-154b-4e46-87e3-2051ccfef4cf.png)

but it is actually the changelog for adapter-auto: https://github.com/sveltejs/kit/blob/master/packages/adapter-auto/CHANGELOG.md#sveltejsadapter-auto

**Comment by dominikg** [comment]
Looks like we have identified what happened. @sveltejs/kit is still in pre-release phase and releases are tagged with `next`. The metadata on the registry seems to follow the `latest` tag and the last release of kit with that tag didn't contain the registry information.
